h2o-core/src/main/java/hex/ModelMojoWriter.java

```diff
@@ -148,7 +148,7 @@ protected final void finishWritingTextFile() throws IOException {
 
   private void addCommonModelInfo() throws IOException {
     int n_categoricals = 0;
-    for (String[] domain : model._output._domains)
+    for (String[] domain : model.scoringDomains())
```

Comment:
The domain {"0","1"} is only inserted if the data had binary column (i.e. no domain) response.
It is because logistic regression can run with binary response ( I did not want to force users to turn it into a factor, other tools don't do it) but also is treated as a classifier. 
Other code expects it to have domain, there was a problem with easy scoring wrapper as I remember, so I insert {"0","1"} as if user ran as.factor.